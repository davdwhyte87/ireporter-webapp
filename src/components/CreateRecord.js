import React from 'react'
import { getRecords, createRecord } from '../actions/recordActions'
import { Link, withRouter } from 'react-router-dom';
import { connect } from 'react-redux'
import propTypes from 'prop-types'
import Alert from './Alert'

class CreateRecord  extends React.Component {
    state = {
        data: {
          title: '',
          type: '',
          comment: '',
          image:''
        },
      }
    handleImage = async(e) => {
      if (e.target.files && e.target.files[0]) {
        const { data } = { ...this.state };
        let reader = new FileReader();
        reader.onload =  function(e) {
            data['image'] = reader.result
        };
        await reader.readAsDataURL(e.target.files[0]);
        await this.setState({ data })
      }
    }
    handleInputChange = (e) => {
        const { data } = { ...this.state };
        data[e.currentTarget.name] = e.currentTarget.value;
        this.setState({ data });
    }
    handleSubmit = (e) => {
        e.preventDefault()
        const { data } = this.state
        this.props.createRecord(data)
    }
  render() {
      const {success} = this.props
      if(success) {
        this.props.history.push('/records')
      }
    return (
      <section className="container-center create-record">
        <div className="card">
        <Alert message={this.props.errors} success={this.props.isError}  />
            <div className="conatiner">
              <img id="img-preview" />
            </div>
            <div className="upload-btn-wrapper" >
                <button className="btn-primary"><i className="fa fa-fw fa-camera"></i> </button>
                <input type="file" id="editImage" onChange={this.handleImage} />
            </div>
            <form id="form">
                <div id="flash"></div> 
                <input type="file" id="editImage" hidden />
                <div className="input-group">
                    <input type="text" name="title" onChange={this.handleInputChange} className="input" required="" placeholder="Title" />
                </div>
                <div className="input-group">
                    <select className="input" name="type" onChange={this.handleInputChange} >
                        <option value="">Record type</option>
                        <option value="red-flag">Red-flag</option>
                        <option value="intervention">Intervention</option>
                    </select>
                    <i className="fa fa-fw fa-question-circle tooltip">
                        <span className="tooltiptext">
                        A red-flag is an incident linked to curroption eg a public officer talking about curropt deeds. An intervention is a call to government agencies
                        eg. bad roads, drug abuse, bad traffic.
                        </span>
                    </i>
                </div>
                <div className="inputBox">
                    <textarea id="mytextarea" onChange={this.handleInputChange} className="input-comment" name="comment" placeholder="Content here.."></textarea>
                </div>
                {/* <div className="input-group">
                    <button className="btn-primary">Get geolocation</button>
                </div> */}
                <div className="input-group">
                    <p id="geolocation-display"></p>
                </div>
                <div className="container-center">
                    <button id="create_btn" type="submit" onClick={this.handleSubmit}  className="btn-primary">{ this.props.loading? 'loading...': 'Create'}<i className="fa fa-fw fa-plus-square"></i></button>
                </div>
            </form>
        </div>
      </section>
    )
    
  }
}
CreateRecord.propTypes = {
    createRecord: propTypes.func.isRequired
  }
  
  CreateRecord.defaultValues = {
    loading: false
  }
  const mapStateToProps = state => {
    return {
      errors: state.recordsReducer.errors,
      loading: state.recordsReducer.loading,
      success: state.recordsReducer.success,
      records: state.recordsReducer.records
    }
  }
  
  export default connect(() => mapStateToProps, { createRecord })(withRouter(CreateRecord))