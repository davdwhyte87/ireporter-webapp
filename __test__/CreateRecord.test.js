import React from 'react';
import { mount, shallow } from 'enzyme';
import CreateRecord from '../src/components/CreateRecord';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';


describe('Create record test', () => {
  const mockStore = configureStore([thunk]);
  const store = mockStore({
    recordsReducer: {
      records: []
    }
  });
  it('should render without crashing', () => {
    const wrapper = mount(
      <Provider store={store}>
      <BrowserRouter>
        <CreateRecord />
      </BrowserRouter>
    </Provider>
    );
    expect(wrapper.find('.create-record').exists()).toBe(true);
    expect(wrapper.find('.container-center').exists()).toBe(true);
  });
  it('should return component', () => {
    const wrapper = mount(
      <Provider store={store}>
      <BrowserRouter>
        <CreateRecord />
      </BrowserRouter>
    </Provider>
    );
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  })
});