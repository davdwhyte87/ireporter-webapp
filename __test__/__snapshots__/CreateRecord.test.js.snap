// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create record test should return component 1`] = `
<section
  className="container-center create-record"
>
  <div
    className="card"
  >
    
    <div
      className="conatiner"
    >
      <img
        id="img-preview"
      />
    </div>
    <div
      className="upload-btn-wrapper"
    >
      <button
        className="btn-primary"
      >
        <i
          className="fa fa-fw fa-camera"
        />
         
      </button>
      <input
        id="editImage"
        onChange={[Function]}
        type="file"
      />
    </div>
    <form
      id="form"
    >
      <div
        id="flash"
      />
      <input
        hidden={true}
        id="editImage"
        type="file"
      />
      <div
        className="input-group"
      >
        <input
          className="input"
          name="title"
          onChange={[Function]}
          placeholder="Title"
          required=""
          type="text"
        />
      </div>
      <div
        className="input-group"
      >
        <select
          className="input"
          name="type"
          onChange={[Function]}
        >
          <option
            value=""
          >
            Record type
          </option>
          <option
            value="red-flag"
          >
            Red-flag
          </option>
          <option
            value="intervention"
          >
            Intervention
          </option>
        </select>
        <i
          className="fa fa-fw fa-question-circle tooltip"
        >
          <span
            className="tooltiptext"
          >
            A red-flag is an incident linked to curroption eg a public officer talking about curropt deeds. An intervention is a call to government agencies eg. bad roads, drug abuse, bad traffic.
          </span>
        </i>
      </div>
      <div
        className="inputBox"
      >
        <textarea
          className="input-comment"
          id="mytextarea"
          name="comment"
          onChange={[Function]}
          placeholder="Content here.."
        />
      </div>
      <div
        className="input-group"
      >
        <p
          id="geolocation-display"
        />
      </div>
      <div
        className="container-center"
      >
        <button
          className="btn-primary"
          id="create_btn"
          onClick={[Function]}
          type="submit"
        >
          Create
          <i
            className="fa fa-fw fa-plus-square"
          />
        </button>
      </div>
    </form>
  </div>
</section>
`;
